### PR TITLE
Multi-forward functionality #1492

### LINKED ISSUE #1492: Implement multi-forward functionality
In Telegram one can forward a message to multiple contacts/chats by long tap and then selecting those. Implement similar multi-forward functionality in Webogram.

### PR DESCRIPTION
**Pull request by azzimuth**
Adds multi-forward functionality. Resolves #1492.

### CODE REVIEW

File: app/js/controllers.js [hunk 1]
```diff
@@ -1941,15 +1941,34 @@ angular.module('myApp.controllers', ['myApp.i18n'])
...
-          selectedCancel()
-          $rootScope.$broadcast('history_focus', {
-            peerString: peerString,
-            attachment: {
-              _: 'fwd_messages',
-              id: selectedMessageIDs
-            }
-          })
+        PeersSelectService.selectPeer({canSend: true}).then(function (peerStrings) {          
+          if(Array.isArray(peerStrings) && peerStrings.length > 1) {
```
Comment (zhukov) [inline app/js/controllers.js]: Space after if

File: app/js/controllers.js [hunk 2]
```diff
@@ -1941,15 +1941,34 @@ angular.module('myApp.controllers', ['myApp.i18n'])
...
+          } else {            
+            $rootScope.$broadcast('history_focus', {
+              peerString: peerStrings,
+              attachment: {
+                _: 'fwd_messages',
+                id: selectedMessageIDs
+              }
+            })
+          }
+          selectedCancel()
```
Comment (zhukov) [inline app/js/controllers.js]: Better move selectedCancel() before `if` statement as was before.

File: app/js/controllers.js [hunk 3]
```diff
@@ -4920,6 +4939,29 @@ angular.module('myApp.controllers', ['myApp.i18n'])
       })
     }
 
+    $scope.dialogMultiSelect = function(peerString){
```
Comment (zhukov) [inline app/js/controllers.js]: space between `){` and also pass `$event` inside this function.

File: app/partials/desktop/dialog.html [hunk 4]
```diff
@@ -18,6 +18,8 @@
     ></i>
   </div>
 
+  <i class="icon icon-select-tick" ng-class="{dialog_selected: isSelected(dialogMessage.peerString)}" ng-mousedown="dialogMultiSelect(dialogMessage.peerString); $event.stopPropagation();"></i>
```
Comment (zhukov) [inline app/partials/desktop/dialog.html]: Don't stop propagation here, pass event to `dialogMultiSelect` as 2nd param.

File: app/js/controllers.js [hunk 5]
```diff
@@ -4920,6 +4939,29 @@ angular.module('myApp.controllers', ['myApp.i18n'])
...
+        $scope.selectedCount--
+        var pos = $scope.selectedPeerIDs.indexOf(peerID)
+        if (pos >= 0) {
+          $scope.selectedPeerIDs.splice(pos, 1)
+        }
+      } else {
+        $scope.selectedPeers[peerID] = AppPeersManager.getPeer(peerID)
+        $scope.selectedCount++
+        $scope.selectedPeerIDs.unshift(peerID)
+      }
```
Comment (zhukov) [inline app/js/controllers.js]: `cancelEvent(event)` here.

**Comment by azzimuth** [comment]
@zhukov I have now implemented the changes suggested in your review.

**Comment by zhukov** [comment]
good job! thank for the PR

**Pull request merged**